lib/httpDigest.js

```diff
@@ -19,13 +22,46 @@ async function create({data, algorithm = 'sha256', useMultihash = true}) {
   if(typeof data !== 'string') {
     data = JSON.stringify(data);
   }
-  const encodedData = new TextEncoder().encode(data);
+  algorithm = algorithm.toLowerCase();
+  const digest = await _getDigest({data, algorithm});
+  if(useMultihash) {
+    return _createMultiHash({digest});
+  }
+  return `SHA-256=${base64.encode(digest)}`;
+}
+
+/**
+ * Verifies digest of data created by create().
+ *
+ * @param {object} options - The options to use.
+ * @param {string|object} options.data - The data to be verified.
+ * @param {string} options.header - The digest header string to
+ *   be verified against.
+ *
+ * @returns {object} - Returns an object with verified true or false.
+ */
+async function verify({data, header}) {
   let digest;
+  if(header.startsWith('SHA-256')) {
```

Comment:
Header keys are case insensitive. We need to parse the key and value here, convert the key to lowercase as before and compare. Then, we'll want to compare the generated, encoded, digest value to the parsed value -- not the entire header. I get that there's an effort to reuse as much code as we can here, but some tweaks are still needed or this (and the comparison below) won't function properly. +1 for figuring out how to do that in the most DRY way.

Reply:
Possible suggestion would be to have a helper that returns the digest as a key + value pairing ... and just have `create` join that with `=` ... and call the helper within `verify` without joining, then you can compare the values only.